warehouse/query-core/src/main/java/datawave/query/tables/keyword/StatefulKeywordUUIDChainStrategy.java

```diff
@@ -15,27 +16,39 @@
 
 import com.google.common.base.Preconditions;
 
+import datawave.core.query.logic.QueryLogic;
 import datawave.microservice.query.Query;
 import datawave.microservice.query.QueryImpl;
 import datawave.query.DocumentSerialization;
 import datawave.query.attributes.Attribute;
 import datawave.query.attributes.Document;
 import datawave.query.function.deserializer.DocumentDeserializer;
 import datawave.query.tables.chained.strategy.FullChainStrategy;
+import datawave.query.tables.keyword.extractor.TagCloudInputExtractor;
+import datawave.query.tables.keyword.extractor.TagCloudInputExtractorException;
+import datawave.util.keyword.TagCloudPartition;
 
 /**
  * Strategy for chaining UUID lookup and keyword extraction queries together. Grabs the results of the lookupUUID query and uses these to generate the query
  * terms that are fed into a keyword extraction query.
  */
 public class StatefulKeywordUUIDChainStrategy extends FullChainStrategy<Entry<Key,Value>,Entry<Key,Value>> {
-
     private static final Logger log = Logger.getLogger(StatefulKeywordUUIDChainStrategy.class);
 
     private int batchSize = -1;
     protected DocumentDeserializer deserializer;
-
-    public StatefulKeywordUUIDChainStrategy(Query settings) {
+    private final QueryLogic<Entry<Key,Value>> nextLogic;
+    // configured extractors to use on the data from nextLogic
+    private final List<TagCloudInputExtractor> extractors;
+    // will be true when a keyword query should be run, false otherwise
+    private final boolean runKeywordQuery;
```

Comment:
Slightly different naming and additional commentary for clarity.
```suggestion
    // will be true when extracting keywords from content, false if the tag cloud input can be satisfied from
    // event metadata.
    private final boolean extractContentKeywords;
```